crud/src/test/java/com/redhat/lightblue/rest/crud/ITCaseCrudResourceTest.java

```diff
@@ -413,8 +452,21 @@ public void testListSavedSearch() throws IOException, ClassNotFoundException, No
 
         // get saved search
         String result = cutCrudResource.getSearchesForEntity("country",null,null).getEntity().toString();
-        Assert.assertTrue(result.indexOf("\"matchCount\":1")!=-1);
+        assertTrue(result.indexOf("\"matchCount\":1")!=-1);
         System.out.println("result:" + result);
 
     }
+
+    @Test
+    @RunAsClient
+    public void testHealthCheck(@ArquillianResource URL url) throws Exception {
+        ClientRequest request = new ClientRequest(UriBuilder.fromUri(url.toURI())
+                .path("healthcheck")
+                .build()
+                .toString());
+        request.accept(MediaType.APPLICATION_JSON);
+        ClientResponse<String> response = request.get(String.class);
+        ObjectNode jsonNode = (ObjectNode) new ObjectMapper().readTree(response.getEntity());
+        assertTrue(jsonNode.elements().next().get("healthy").asBoolean());
+    }
```

Comment:
@sameer31 ^ Here is the basic healthcheck test.

We would want to add one of these to the cert auth project, and make sure it discovers the LDAP role provider to add a health check for that, too. We could also consider testing how we wire up the role provider health checks independently I guess.